Mathlib/Probability/Distributions/Exponential.lean

```diff
@@ -0,0 +1,251 @@
+noncomputable
+def expMeasure (r : ℝ) : Measure ℝ :=
+  volume.withDensity (exponentialPdf r)
+
+instance instIsProbabilityMeasureExponential (r : ℝ) [Fact (0 < r)] :
+    IsProbabilityMeasure (expMeasure r) where
+  measure_univ := by simp [expMeasure, lintegral_exponentialPdf_eq_one r Fact.out]
+
+section ExponentialCdf
+
+/-- CDF of the exponential Distribution -/
+noncomputable
+def exponentialCdfReal (r : ℝ) : StieltjesFunction :=
+    cdf (expMeasure r)
+
+lemma expCdf_eq_integral (r : ℝ) [Fact (0 < r)] : exponentialCdfReal r
+    = fun x ↦ ∫ x in Iic x, exponentialPdfReal r x := by
+  ext x
+  rw [exponentialCdfReal,cdf_eq_toReal]
+  simp only [expMeasure, measurableSet_Iic, withDensity_apply]
+  rw [integral_eq_lintegral_of_nonneg_ae]; exact rfl
+  · exact ae_of_all _ fun a ↦ by simp [Pi.zero_apply, exponentialPdfReal_nonneg Fact.out a]
+  · exact (Measurable.aestronglyMeasurable (measurable_exponentialPdfReal r)).restrict
+
+lemma expCdf_eq_lintegral (r : ℝ) [Fact (0 < r)] : exponentialCdfReal r =
+    fun x ↦ ENNReal.toReal (∫⁻ x in Iic x, exponentialPdf r x) := by
+  ext x
+  simp only [exponentialPdf, exponentialCdfReal, cdf_eq_toReal]
+  simp only [expMeasure, measurableSet_Iic, withDensity_apply]
+  rfl
+
+open Topology
+
+lemma antiDeriv_expDeriv_pos {r x : ℝ} :
+    HasDerivAt (fun a ↦ -1 * exp (-(r * a))) (r * exp (-(r * x))) x := by
+  convert (((hasDerivAt_id x).const_mul (-r)).exp.const_mul (-1)) using 1
+  · simp only [id_eq, neg_mul]
+  simp only [id_eq, neg_mul, mul_one, mul_neg, one_mul, neg_neg, mul_comm]
+
+lemma lint_eq_antiDeriv (r : ℝ) (hr : 0 < r) : ∀ x : ℝ,
+    (∫⁻ y in Iic x, exponentialPdf r y =
+    ENNReal.ofReal (if 0 ≤ x then 1 - exp (-(r * x)) else 0)) := by
+  intro x'
+  split_ifs with h
+  case neg =>
+    simp only [exponentialPdf_eq]
+    rw [set_lintegral_congr_fun measurableSet_Iic, lintegral_zero, ENNReal.ofReal_zero]
+    exact ae_of_all _ fun a (_ : a ≤ _) ↦ by rw [if_neg (by linarith), ENNReal.ofReal_eq_zero]
+  case pos =>
+    rw [lintegral_split_bounded _ h, lintegral_nonpos (le_refl 0), zero_add]
+    simp only [exponentialPdf_eq]
+    rw[set_lintegral_congr_fun measurableSet_Icc (ae_of_all _
+        (by intro a ⟨(hle : _ ≤ a), _⟩; rw [if_pos hle]))]
+    rw [←ENNReal.toReal_eq_toReal _ ENNReal.ofReal_ne_top, ←integral_eq_lintegral_of_nonneg_ae
+        (eventually_of_forall fun _ ↦ le_of_lt (mul_pos hr (exp_pos _)))]
+    have : ∫ a in uIoc 0 x', r * rexp (-(r * a)) = ∫ a in (0)..x', r * rexp (-(r * a)) := by
+      rw [intervalIntegral.intervalIntegral_eq_integral_uIoc, smul_eq_mul, if_pos h, one_mul]
+    rw [integral_Icc_eq_integral_Ioc, ← uIoc_of_le h, this]
+    rw [intervalIntegral.integral_eq_sub_of_hasDeriv_right_of_le h
+      (f := fun a ↦ -1 * rexp (-(r * a))) _ _]
+    rw [ENNReal.toReal_ofReal_eq_iff.2 (by norm_num; positivity)]
+    · norm_num; ring
+    · simp only [intervalIntegrable_iff, uIoc_of_le h]
+      apply Integrable.const_mul (exp_neg_integrableOn_Ioc hr)
```

Comment:
When `apply` closes a goal, it's better for readability of the proof to use `exact` instead:
```suggestion
      exact Integrable.const_mul (exp_neg_integrableOn_Ioc hr) _
```